tools/custom-stats-qc/1.0.0/custom-stats-qc__1.0.0.cwl

```diff
@@ -0,0 +1,154 @@
+cwlVersion: v1.1
+class: CommandLineTool
+
+# Extensions
+$namespaces:
+    s: https://schema.org/
+    ilmn-tes: http://platform.illumina.com/rdf/ica/
+$schemas:
+  - https://schema.org/version/latest/schemaorg-current-http.rdf
+
+# Metadata
+s:author:
+    class: s:Person
+    s:name: Sehrish Kanwal
+    s:email: sehrish.kanwal@umccr.org
+
+# ID/Docs
+id: custom-stats-qc--1.0.0
+label: custom-stats-qc v(1.0.0)
+doc: |
+    A tool to extract custom QC metrics from samtools stats output and convert to json format.
+
+# ILMN Resources Guide: https://support-docs.illumina.com/SW/ICA/Content/SW/ICA/RequestResources.htm
+hints:
+    ResourceRequirement:
+        ilmn-tes:resources:
+            tier: standard
+            type: standard
+            size: small
+        coresMin: 1
+        ramMin: 4000
+    DockerRequirement:
+        dockerPull: quay.io/umccr/alpine-jq:1.6
+
+requirements:
+  InlineJavascriptRequirement: {}
+  InitialWorkDirRequirement:
+    listing:
+      - entryname: "run-custom-qc.sh"
+        entry: |
+          #!/usr/bin/env sh
+
+          # Set up for failure
+          set -xeuo pipefail
+
+          # Sample metadata
+          JSON_STRING=\$( jq -n \\
+                            --arg sm "$(inputs.sample_id)" \\
+                            '{sampleID: $sm}' )
+          echo "$(inputs.output_json_filename).json"
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract average base quality
+          avg_base_qual=\$(grep -h "SN	average quality" "$(inputs.output_samtools_stats.path)" | cut -f 3)
```

Comment:
The samtools stats file has this hint at the top of the file:
```
# Summary Numbers. Use `grep ^SN | cut -f 2-` to extract this part.
```

Given all of our queries are in the Summary numbers section it may be cleaner to extract this first to a temp file.  

```
samtools_stats_file="$(runtime.tmpdir)/summary_numbers.txt" 
grep ^SN "$(inputs.output_samtools_stats.path)" | cut -f 2- > "${samtools_stats_file}"
```

We can then use `grep "query" "${samtools_stats_file}" | cut -f 2` to pull out values.

Reply:
Updated https://github.com/umccr/cwl-ica/blob/52ee1dd285bbb8d113e4e53f9dbf23c38cb9c750/tools/custom-stats-qc/1.0.1/custom-stats-qc__1.0.1.cwl#L66